doc/ssdmond/ssdmond-hld.md

```diff
@@ -0,0 +1,129 @@
+
+**Reserved Blocks Count** - Reserving some number of filesystem blocks for use by privileged processes is done to avoid filesystem fragmentation, and to allow system daemons, such as syslogd(8), to continue to function correctly after non-privileged processes are prevented from writing to the filesystem. Normally, the default percentage of reserved blocks is 5%.<sup>[1](#1-man-tune2fs)</sup>
+
+**Temperature** - Extreme temperatures can affect SSD performance. Excessive heat can lead to throttling to prevent damage, while extreme cold can slow down data access.
+
+
+### **2.2 Priority 1 Attributes**
+
+**These are a combination of static (S) and dynamic (D) fields, offering secondary information that provides additional context about the SSD**
+
+    - Vendor Model (S)
+    - Serial Number (S)
+    - Firmware (S)
+    - Health (D)
+
+These fields are self-explanatory.
+
+
+### **2.3 `ssdmond` Daemon Flow**
+
+0. Vendor would be responsible for configuring the following values:
+    - **loop timeout** - This determines how often the dynamic information would be updated. Default is 6 hours.
+    - **SSD vendor-specific search terms** - This would ensure that all the attributes are properly parsed from the device.
+
+1. `ssdmond` would be started by the `pmon` docker container
+2. The daemon would gather the static info once init-ed, by leveraging the `ssdutil` utility and update the StateDB
+3. It would periodically parse the priority 0 attributes either by leveraging `ssdutil` or directly through Linux utilities and update the StateDB.
+
+This is detailed in the sequence diagram below:
+
+![image.png](images/SSDMOND_SequenceDiagram.png)
+
+
+NOTE: While it is previously established that we use the abstraction provided by the `ssdutil` class to offer vendors the opportunity to implement their own SSD parsing logic, the [primary intent](#1-overview) of this iteration of the design is to enable streaming telemetry of these attributes, i.e., update the StateDB with the parsed data. A design choice is therefore made to bypass the abstraction logic in favor of this goal in the interim, with the expressed understanding that said abstraction will follow in a [future version](#future-work) of this daemon.
+
+
+### **2.4 Data Collection Logic**
+
+The SONiC OS already contains logic to parse information about SSDs from several vendors by way of the `ssdutil` platform utility. We leverage this utility to gather the following information:
+
+- Priority 0: Temperature
+- Priority 1: All aforementioned attributes
+
+This section will therefore only go into detail about data collection of attributes mentioned in [section 2.1](#21-priority-0-attributes):
+
+#### **2.4.1 IO Reads/Writes**
+
+- grep `/proc/diskstats` for statistics about the SSD of interest
```

Comment:
@assrinivasan Not sure how these stats are preserved across reboots. The write stats should account for entire lifetime the SSD has been powered ON.

Reply:
Addressed in latest commit